src/ni/datastore/_client.py

```diff
@@ -510,10 +511,31 @@ def query_test_adapters(self, odata_query: str) -> Iterable[TestAdapter]:
             TestAdapter.from_protobuf(test_adapter) for test_adapter in query_response.test_adapters
         ]
 
-    # TODO: Also support providing a file path?
-    def register_schema(self, schema: str) -> str:
-        """Register a schema in the metadata store."""
-        register_request = RegisterSchemaRequest(schema=schema)
+    def register_schema_from_file(self, schema_file_path: Path | str) -> str:
+        """Register a schema obtained from the specified file in the metadata store.
+
+        Args:
+            schema_file_path: The path at which the schema file is located
+
+        Raises:
+            FileNotFoundError: If the schema file does not exist.
+        """
+        if isinstance(schema_file_path, str):
+            schema_file_path = Path(schema_file_path)
+
+        if not schema_file_path.exists():
+            raise FileNotFoundError(f"Schema file not found: {schema_file_path}")
+
+        schema_contents = schema_file_path.read_text()
```

Comment:
Specify encoding when reading contents of file.
https://github.com/ni/ni-apis-python/blob/main/packages/ni.measurementlink.pinmap.v1.client/src/ni/measurementlink/pinmap/v1/client/_client.py#L61